numpy/core/include/numpy/npy_3kcompat.h

```diff
@@ -221,33 +221,21 @@ npy_PyFile_Dup2(PyObject *file, char *mode, npy_off_t *orig_pos)
                         "Getting a FILE* from a Python file object failed");
     }
 
+    ret = PyObject_CallMethod(file, "seekable", "");
+    seekable = ret == Py_True;
+    Py_DECREF(ret);
```

Comment:
This needs to handle exceptions from `file.seekable` - as is, it segfaults if that happens.

Reply:
Thanks, I think i got it?

Reply:
There's no need to set your own error message - better to just let the error propagate, I think

Reply:
Is it weird that the statement above sets the error but does not return immediately?

Reply:
Should I also add a comment that this is an error? Or is it assume that numpy contributors will know that these special values constitute errors?